### ISSUE TITLE
Editing a user doesn't have constrains for the admin user

### ISSUE THREAD

**Issue by skateman** [ManageIQ/manageiq-api#974]
In the UI the user editing form (and controller) doesn't allow the user to change the name or the assigned groups of the `admin` user. However, this is not the case in the API and it is possible to rename the `admin` user or assign it to certain groups which I think might break RBAC stuff.

Ideally these constraints should be in the DB model, but we don't really do that because that might hurt db seeding so I guess we should mimic the behavior of the UI in the API.

@miq-bot assign @gtanzillo 
@miq-bot add_label bug
cc @Fryguy

**Comment by Fryguy** [comment]
Can you link to specific code...I agree it should move to the model, actually, but it depends on the specifics

**Comment by skateman** [comment]
![Screenshot from 2020-12-07 15-16-26](https://user-images.githubusercontent.com/649130/101361393-3419e580-389f-11eb-8b3c-650c1cea33ce.png)

https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/views/ops/_rbac_user_details.html.haml#L9

**Comment by kbrock** [comment]
For this case, putting in the model may make sense.
But we're hitting more and more use cases where we have business logic in the controllers and no good place to duplicate this logic in the api controllers.

I'm ok making a slew of changes across all tests to avoid the env.test?
specifically with users and requiring tenants and groups, we've made quite a few changes to make them behave in tests in a non-one off manner

It would be nice if people could keep in mind that we need a more permanent recipe for where we want to put this logic

**Comment by skateman** [comment]
> It would be nice if we could keep in mind that we need a more permanent recipe for where we want to put this controller logic in a more sustainable place.

These are more like constrains that should be in a model, at least that's where I'd put them in my own hypothetical Rails app :thinking:

**Comment by Fryguy** [comment]
@skateman I agree.  Generally, constraints that apply to all users should be in the model (or even the database)...things like specific value ranges, enums, certain validations.  _Some_ things we can't put in the model, particular if they only apply to a subset of users, or if they prevent standard backend usage like deleting records.

In this particular case, while this is a user-specific change, that user in question is well-defined, and I think the model is an appropriate place.
